### ISSUE TITLE
Pooling throws an exception in Tegra TX1

### ISSUE THREAD

**Issue by dimatura** [pytorch/pytorch#1906]
Hi all, great work with pytorch. I'm trying to use pytorch in a Jetson TX1, but running into some issues. One of them is pretty simple: pooling doesn't seem to work. 

First, some details:
- Jetson TX1, a 64-bit ARM architecture with a tegra TX1
- Ubuntu 16.04 aarch64
- CUDA 8.0, CUDNN 5105
- python 2.7
- Built from source (conda not available yet) from at least two git commits - 7c24a, Jun 23 and one from roughly a week earlier. ~~Currently compiling the 0.1.12 tag - will update when done.~~ Tag 1.12 (ccdf54) actually works correctly, with and without cuda.

Here's a simple example:
```python
import torch
import torch.nn as nn                                                                                                                                                                                              
from torch.autograd import Variable                                                                                                                                                                                
                                                                                                                                                                                                                    
net = nn.MaxPool2d(2)#.cuda()                                                                                                                                                                                      
#net = nn.AvgPool2d(2)#.cuda()                                                                                                                                                                                      
                                                                                                                                                                                                                    
xd = torch.FloatTensor(1, 3, 448, 448)                                                                                                                                                                           
x = Variable(xd, volatile=True)#.cuda()                                      `                                                                                                                                      
y = net.forward(x)                                                                                                                                                                                                 
print(y.data.mean())
```
when using MaxPool2d, I get 
```
terminate called after throwing an instance of 'THArgException'
  what():  can't expand an empty tensor at /home/ubuntu/apps/pytorch/torch/lib/TH/generic/THTensor.c:295
Aborted
```

~~when using AvgPool2d, oddly I get a different error:~~  AvgPool2d works as expected.

Since I'm not too experienced with the internals, I'm not sure what changes matter, but I changed a few things (using cuda or cpu, volatile or not, the tensor size, the pooling parameters) and always got the same result. As a very very approximate workaround I've been using extra striding in my convolutions, but obviously that's not ideal ;).

**Comment by dimatura** [comment]
Wait - I am closing this one. Really stupid error on my part - (1, 3, 448, 448) should *not* be a tuple.

**Comment by dimatura** [comment]
Wait, I spoke too soon - `AvgPool2d` works fine, `MaxPool2d` is still broken.

**Comment by dimatura** [comment]
Tag 1.12 (ccdf54) works, so this seems to be a regression.

**Comment by soumith** [comment]
this is a weird error. Are you sure CuDNN is being detected at compile time?

You might have to set `CUDNN_INCLUDE_DIR` and `CUDNN_LIB_DIR` environment variables.
You can check cudnn's status with: `print(torch.backends.cudnn.enabled)`

**Comment by dimatura** [comment]
Agreed, it is weird - inspection of the source code didn't reveal any obvious issue (though I'm not really familiar with torch internals). 

Is there anything in the error suggesting cudnn is not detected? For what it's worth, I am fairly confident it was detected, as `torch.backends.cudnn.enabled` returned true. I wanted cudnn since I want to squeeze as much performance out as possible of this thing, and I had to compile the source code a few times before getting it right (in effect, I did have to set up the env vars at compile time).
